src/ucs/config/parser.c

```diff
@@ -1175,25 +1175,38 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
                                          int ignore_errors)
 {
     ucs_status_t status;
-    char prefix[128];
+    int cstm_prefix_len;
+    int full_prefix_len;
 
     /* Set default values */
     status = ucs_config_parser_set_default_values(opts, fields);
+    ucs_assert(env_prefix != NULL);
     if (status != UCS_OK) {
         goto err;
     }
 
+    /* Find if env_prefix consists of base prefix and custom prefix, base prefix
+     * should not contain underscores "_" except at the very end
+     * e.g in env_prefix = "LONG_PREFIX_UCX_LOG_LEVEL":
+     * "LONG_PREFIX_" is custom_prefix
+     * "UCX_"         is base prefix */
+    full_prefix_len = strlen(env_prefix);
+    cstm_prefix_len = full_prefix_len - 2;
+    while ((cstm_prefix_len >= 0) && (env_prefix[cstm_prefix_len] != '_')) {
+        cstm_prefix_len -= 1;
+    }
+    cstm_prefix_len += 1;
```

Comment:
maybe add helper function? `const char *ucs_config_parser_get_sub_prefix(const char *env_prefix)`

let's call "custom prefix" a "sub-prefix" since "custom" is confusing - the whole prefix is "custom" since this function accepts it as input argument